examples/src/examples/graphics/render-asset.mjs

```diff
@@ -0,0 +1,118 @@
+    };
+
+    const device = await pc.createGraphicsDevice(canvas, gfxOptions);
+    const createOptions = new pc.AppOptions();
+    createOptions.graphicsDevice = device;
+
+    createOptions.componentSystems = [
+        // @ts-ignore
+        pc.RenderComponentSystem,
+        // @ts-ignore
+        pc.CameraComponentSystem,
+        // @ts-ignore
+        pc.LightComponentSystem
+    ];
+    createOptions.resourceHandlers = [
+        // @ts-ignore
+        pc.TextureHandler,
+        // @ts-ignore
+        pc.ContainerHandler
+    ];
+
+    const app = new pc.AppBase(canvas);
+    app.init(createOptions);
+
+    // Set the canvas to fill the window and automatically change resolution to be the same as the canvas size
+    app.setCanvasFillMode(pc.FILLMODE_FILL_WINDOW);
+    app.setCanvasResolution(pc.RESOLUTION_AUTO);
+
+    const assetListLoader = new pc.AssetListLoader(Object.values(assets), app.assets);
+    assetListLoader.load(() => {
+
+        app.start();
+
+        /** @type {pc.Entity[]} */
+        const cubeEntities = [];
+
+        // get the instance of the cube it set up with render component and add it to scene
+        cubeEntities[0] = assets.cube.resource.instantiateRenderEntity();
+        cubeEntities[0].setLocalPosition(7, 12, 0);
+        cubeEntities[0].setLocalScale(3, 3, 3);
+        app.root.addChild(cubeEntities[0]);
+
+        // clone another copy of it and add it to scene
+        cubeEntities[1] = cubeEntities[0].clone();
+        cubeEntities[1].setLocalPosition(-7, 12, 0);
+        cubeEntities[1].setLocalScale(3, 3, 3);
+        app.root.addChild(cubeEntities[1]);
+
+        // get the instance of the statue and set up with render component
+        const statueEntity = assets.statue.resource.instantiateRenderEntity();
+        app.root.addChild(statueEntity);
+
+        // Create an Entity with a camera component
+        const camera = new pc.Entity();
+        camera.addComponent("camera", {
+            clearColor: new pc.Color(0.2, 0.1, 0.1),
+            farClip: 100
+        });
+        camera.translate(-20, 15, 20);
+        camera.lookAt(0, 7, 0);
+        app.root.addChild(camera);
+
+        // set skybox
+        app.scene.envAtlas = assets.helipad.resource;
+        app.scene.toneMapping = pc.TONEMAP_ACES;
+        app.scene.skyboxMip = 1;
+
+        // spin the meshes
+        app.on("update", function (dt) {
+
+            if (cubeEntities[0]) {
+                cubeEntities[0].rotate(3 * dt, 10 * dt, 6 * dt);
+            }
+
+            if (cubeEntities[1]) {
+                cubeEntities[1].rotate(-7 * dt, 5 * dt, -2 * dt);
+            }
+
+            if (statueEntity) {
+                statueEntity.rotate(0, -12 * dt, 0);
+            }
+
+        });
+    });
+    return app;
+}
+
+export class RenderAssetExample {
+    static CATEGORY = 'Graphics';
+    static NAME = 'Render Asset';
+    static WEBGPU_ENABLED = true;
+    static example = example;
```

Comment:
could we embed the example function here?

Reply:
Yes, that's still possible. I thought about that these classes could be removed entirely to make it simpler to use examples (for standalone .html files later). They don't contain "methods" (aka no access to `this`) and it's just adding a level of intendation. Having it not in there also helps with removal of Babel/Prettier and also the handmade "parsers" (just to remove level of intendation for code reload in Monaco). Another reason is to have a simple toString() which resembles an entire function:

![image](https://github.com/playcanvas/engine/assets/5236548/3e2b03ba-28f6-4d07-8b88-590bea4b1257)

So I pretty much just saw advantages to rewrite the code like that to make the code shorter/smaller/easier to read.